modules/fedimint-meta-server/src/lib.rs

```diff
@@ -279,32 +279,18 @@ impl ServerModule for Meta {
             },
         ) in desired
         {
-            let consensus_value = &Self::get_consensus(dbtx, key).await;
             let consensus_submission_value =
                 Self::get_submission(dbtx, key, self.our_peer_id).await;
+
             if consensus_submission_value.as_ref()
-                == Some(&MetaSubmissionValue {
+                != Some(&MetaSubmissionValue {
                     value: desired_value.clone(),
                     salt,
                 })
             {
-                // our submission is already registered, nothing to do
-            } else if consensus_value.as_ref() == Some(&desired_value) {
```

Comment:
This ignores the current consensus value. After submitted value becomes consensus, the submission becomes `None`. So here, it will no longer `==` making this keep returning the value as a submission consensus item. But the `process_consensus_proposal` will detect that the submission is equal to consensus and ignore it. But since meta module fails to return `Err`... it will keep spamming and making the consensus history grow larger.

Not sure how to even write a test for it, because it's invisible on the API level.

Reply:
Nah, we weren't aware of that cleanup logic. I thought it was an easy win, but it isn't, so let's leave it be for now since it seems to work.